xs2a-gateway-service-api-adapter/src/main/java/de/adorsys/xs2a/gateway/signing/service/hashing/ShaHashingService.java

```diff
@@ -0,0 +1,23 @@
+package de.adorsys.xs2a.gateway.signing.service.hashing;
+
+import de.adorsys.xs2a.gateway.signing.exception.HttpRequestSigningException;
+import de.adorsys.xs2a.gateway.signing.service.algorithm.HashingAlgorithm;
+
+import java.nio.charset.Charset;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public abstract class ShaHashingService implements HashingService {
```

Comment:
Seems it is not `SHA` hashing service, it looks like abstract hashing service

Reply:
Renamed to `AbstractShaHashingService`, thanks! 
Sha is used in a name as this class should be used for SHA algorithms only. I'm not sure, that the same Java API methods will be used for non-SHA algorithms.